### ISSUE TITLE
Button has incorrect 

### ISSUE THREAD

**Issue by swissspidy** [WordPress/gutenberg#34879]
### Description

The `Button` component uses `uniqueId()` for the `aria-describedby` attribute to point to a `VisuallyHidden` element with that ID.

`uniqueId()` produces numeric IDs, however, IDs in HTML must not start with a number.

`uniqueId()` should be used with a prefix (or perhaps replaced by `uuid`) and ideally be memoized so it doesn't change on every render.

https://github.com/WordPress/gutenberg/blob/5cca2e1a8021679aaf2fc713c1a2c1cce168c374/packages/components/src/button/index.js#L142-L145

https://github.com/WordPress/gutenberg/blob/69e348ec80dc2cb9d537a8ba3cf6774b48636461/packages/block-editor/src/components/block-switcher/index.js#L167

### Step-by-step reproduction instructions

Render the `Button` component yourself or observe the wrong ID in the block switcher.

### Screenshots, screen recording, code snippet

![Screenshot 2021-09-16 at 15 13 40](https://user-images.githubusercontent.com/841956/133618985-08ad90ce-e1f5-4497-807f-aa742c64da05.png)

### Environment info

WordPress 5.8.1

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes
